_extras/sequence_alignment.md

```diff
@@ -0,0 +1,516 @@
+
+Note: running the same command will provide an error as `clustalo` will refuse to overwrite the existing `peptides_aligned.fasta` file. 
+This is a good thing! However, it is possible to override this safety by adding **`--force`** to the command. 
+
+`clustalo` can also output the alignment in other formats as detailed in the help under the optional modifier 
+`--outfmt`. We can try the `clustal` format for a more visual output by adding `--outfmt=clu`:
+
+~~~
+$ docker run --rm --mount type=bind,source=${PWD},target=/data pegi3s/clustalomega --outfmt=clu -i /data/peptides.fasta -o /data/peptides_aligned.clu
+$ cat peptides_aligned.clu
+~~~
+{: .language-bash}
+~~~
+CLUSTAL O(1.2.4) multiple sequence alignment
+
+
+GIP           YAEGTFISDYSIAMDKIRQQDFVNWLLAQ----
+GLP-1         HAEGTFTSDVSSYLEGQAAKEFIAWLVKGRG--
+GLP-2         HADGSFSDEMNTILDNLAARDFINWLIQTKITD
+glucagon      HSQGTFTSDYSKYLDSRRAQDFVQWLMNT----
+              :::*:* .: .  ::    ::*: **:      
+~~~
+{: .output}
+
+> ## Challenge: align more sequences.
+> How would you proceed to align the hemogloin and globin sequences contained within the file `hglob.fasta`? located in the `~/Desktop/docker-intro/peptides` you copied?
+> 
+> Would anything be different to align the 32 spike protein sequence from virus SARS_Cov-2 found in file `spike_32.fa` within the `~/Desktop/docker-intro/covid` directory?
+> 
+> > ## Solution
+> >
+> > Since we cannot run (for now) the container interactively we need to provide names for the input file and output result files. We can also select an alternate output format.
+> > ~~~
+> > $ docker run --rm --mount type=bind,source=${PWD},target=/data pegi3s/clustalomega --outfmt=clu -i /data/hglob.fasta -o /data/hglob_aligned.clu
+> > ~~~
+> > {: .language-bash}
+> > For the larger set in `spike_32.fa` we could add `-v` (verbose) to obtain more information from `clustalo` as the software is computing the alignment. 
+> > Nothing else needs to be fundamentally different. Just remember to add `--force` to run the same command again.
+> >  ~~~
+> > $ docker run --rm --mount type=bind,source=${PWD},target=/data pegi3s/clustalomega -v --outfmt=clu -i /data/spike_32.fa -o /data/spike_32_aln.clu
+> > ~~~
+> > {: .language-bash}
+> > ~~~
+> > Using 4 threads
+> > Read 32 sequences (type: Protein) from /data/spike_32.fa
+> > not more sequences (32) than cluster-size (100), turn off mBed
+> > Calculating pairwise ktuple-distances...
+> > Ktuple-distance calculation progress done. CPU time: 0.37u 0.01s 00:00:00.38 Elapsed: 00:00:00
+> > Guide-tree computation done.
+> > Progressive alignment progress: 64 % (20 out of 31
+> > ~~~
+> > {: .output}
+> {: .solution}
+{: .challenge}
+
+> ## Challenge: bypassing the entry point.
+> How would you proceed to run this container interactively
+> since adding `-it` has no effect due to the built-in entry point command?
+> This aspect was explored within an earlier lesson chapter. 
```

Comment:
```suggestion
> This aspect was explored within an earlier episode of this lesson. 
```

Reply:
Might suggest changing this to mention the exact episode?

Reply:
I add link as a **relative** path since the full URL would be lost when the lesson is no longer on the "Beta" format:

This aspect was explored within an earlier episode of this lesson ([Creating More Complex Container Images](../advanced-containers/index.html)).